### ISSUE TITLE
Terminal: Set path to current document menu option

### ISSUE THREAD

**Issue by jeremypw** [elementary/code#735]
I have searched open and closed issues for duplicates.

## Feature
A quick way to cd to the directory containing the current document would be useful.

**Describe the solution you'd like**
Add a menu option "Set path to current document" to the terminal context menu.

**Existing work**
Geany has this feature

**Describe alternatives you've considered**
Smart path following could be an option but is difficult to get right for everyone. Some way of overriding it would still probably be necessary.

**Comment by cybarspace** [comment]
@jeremypw 
Wouldn't it be easier to port this feature from Gedit than from Geany ?
Gedit is quite similar to Code. The terminal plugin in Gedit already has this feature.

**Comment by jeremypw** [comment]
@cybarspace The code required is straightforward enough to write from scratch in Vala but I understand it is mooted to retire the terminal plugin anyway in favour of a more focused build/install feature so it is uncertain whether this feature would be merged.

**Comment by jeremypw** [comment]
@cybarspace Have now pushed a PR providing this feature (#810) although not sure what its chances of being merged are. If you are able to build and install that branch you might like to try it out and comment.

**Comment by cybarspace** [comment]
@jeremypw Sorry, I don't have the know-how to do that. If it makes it's way into stable or the AUR, I might be able to try it. 😅

**Comment by cybarspace** [comment]
> I understand it is mooted to retire the terminal plugin anyway in favour of a more focused build/install feature

I don't see how that can ever replace a whole integrated terminal. A terminal has more functions and purposes than just build/install. 😕

**Comment by jeremypw** [comment]
@cybarspace. No problem.

>I don't see how that can ever replace a whole integrated terminal. A terminal has more functions and purposes than just build/install

Yes, but are those functions and purposes *directly* relevant to a code editor?  If so, then those functions can be added to build/install. Otherwise, there is a separate Terminal app you can use.

**Comment by cybarspace** [comment]
> Yes, but are those functions and purposes _directly_ relevant to a code editor? If so, then those functions can be added to build/install. Otherwise, there is a separate Terminal app you can use.
Most industry-grade code editors have a built-in terminal. It's inconvenient not to have one.
But if the devs wish to drop it, I won't argue...
Just integrate Git properly first, then remove terminal. Otherwise VS Code it is.

**Comment by jeremypw** [comment]
Agreed.  Good git support is one of the features needed to make the terminal plugin redundant.

**Comment by cybarspace** [comment]
@jeremypw Also, please make it possible to spawn REPLs of languages like Python, Clojure, Ruby, etc. or edit the existing Python extension such that it'd be easily reprogrammable to support these and other languages, like we could make spin-off add-ons based on the Python one for each language we wish to support. As we know, the REPL is very important for Clojure devs (that's why many use Emacs, or Calva in VS Code).

**Comment by jeremypw** [comment]
@cybarspace  Thanks for the suggestion.  Could you raise it as a separate issue please.  Maybe I'll create a Project for superceding the Terminal plugin and related issues (i.e. features that need to be replicated in the main UI first) can be linked to it.

**Comment by jeremypw** [comment]
I am going to close this as you can change the Terminal working directory as required using the sidebar context menu on the relevant folder item.  Moreover I now rarely if ever need this function.
